aea/crypto/cosmos.py

```diff
@@ -0,0 +1,296 @@
+        """
+        Return a public key in hex format.
+
+        :return: a public key string in hex format
+        """
+        return self._public_key
+
+    @property
+    def address(self) -> str:
+        """
+        Return the address for the key pair.
+
+        :return: a display_address str
+        """
+        return self._address
+
+    def _load_private_key_from_path(self, file_name) -> SigningKey:
+        """
+        Load a private key in hex format from a file.
+
+        :param file_name: the path to the hex file.
+        :return: the Entity.
+        """
+        path = Path(file_name)
+        try:
+            if path.is_file():
+                with open(path, "r") as key:
+                    data = key.read()
+                    signing_key = SigningKey.from_string(
+                        bytes.fromhex(data), curve=SECP256k1
+                    )
+            else:
+                signing_key = self._generate_private_key()
+            return signing_key
+        except IOError as e:  # pragma: no cover
+            logger.exception(str(e))
+
+    def sign_message(self, message: bytes, is_deprecated_mode: bool = False) -> str:
+        """
+        Sign a message in bytes string form.
+
+        :param message: the message to be signed
+        :param is_deprecated_mode: if the deprecated signing is used
+        :return: signature of the message in string form
+        """
+        signature_compact = self.entity.sign_deterministic(
+            message, hashfunc=hashlib.sha256, sigencode=sigencode_string_canonize,
+        )
+        signature_base64_str = base64.b64encode(signature_compact).decode("utf-8")
+        return signature_base64_str
+
+    def sign_transaction(self, transaction: Any) -> Any:
+        """
+        Sign a transaction in bytes string form.
+
+        :param transaction: the transaction to be signed
+        :return: signed transaction
+        """
+        message_str = json.dumps(transaction, separators=(",", ":"), sort_keys=True)
+        message_bytes = message_str.encode("utf-8")
+        signed_transaction = self.sign_message(message_bytes)
+        return signed_transaction
+
+    def recover_message(
+        self, message: bytes, signature: str, is_deprecated_mode: bool = False
+    ) -> Address:
+        """
+        Recover the address from the hash.
+
+        :param message: the message we expect
+        :param signature: the transaction signature
+        :param is_deprecated_mode: if the deprecated signing was used
+        :return: the recovered address
+        """
+        signature_b64 = base64.b64decode(signature)
+        verifying_keys = VerifyingKey.from_public_key_recovery(
+            signature_b64, message, SECP256k1, hashfunc=hashlib.sha256,
+        )
+        # TODO: pick correct key
+        public_key = verifying_keys[0].to_string("compressed").hex()
+        address = self.get_address_from_public_key(public_key)
+        return address
```

Comment:
@steevi, we are trying to recover the address from the signature and message. In ethereum this works, but here there is some indeterminacy. Can you advice?